fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeActivity.java

```diff
@@ -66,61 +58,58 @@ protected void onCreate(Bundle savedInstanceState) {
+    navbar.setOnItemSelectedListener(
+        item -> {
+          HomeTab tab = HomeTab.findByMenu(item.getItemId());
+          boolean selected = openTab(tab);
+          if (selected) viewModel.setCurrentTab(tab);
+          return selected;
+        });
+    // Set an empty reselect listener to disable the onSelectListener when pressing multiple times
+    navbar.setOnItemReselectedListener(item -> {});
+  }
 
-    MenuItem connectItem = navbar.getMenu().getItem(CONNECT_POSITION);
-    MenuItem launchItem = navbar.getMenu().getItem(LAUNCH_POSITION);
+  /** Setup the listeners that changes the navigation bar menus */
+  private void setupNavigationBarListener() {
+    MenuItem connectItem = navbar.getMenu().getItem(HomeTab.CONNECT.ordinal());
+    MenuItem launchItem = navbar.getMenu().getItem(HomeTab.LAUNCH.ordinal());
+    MenuItem socialMediaItem = navbar.getMenu().getItem(HomeTab.SOCIAL.ordinal());
 
-    mViewModel
+    // Gray out the launch and connect buttons depending on the wallet state
+    viewModel
         .getIsWalletSetUpEvent()
         .observe(
             this,
-            aBoolean -> {
-              // We set transparency of the
-              if (Boolean.TRUE.equals(aBoolean)) {
-                connectItem.setIcon(R.drawable.ic_home_connect_opaque_foreground);
-                launchItem.setIcon(R.drawable.ic_home_launch_opaque_foreground);
-              } else {
-                connectItem.setIcon(R.drawable.ic_home_connect_transparent_foreground);
-                launchItem.setIcon(R.drawable.ic_home_launch_transparent_foreground);
-              }
+            walletSetup -> {
+              // We set the button icon depending on the livedata value
+              boolean setup = Boolean.TRUE.equals(walletSetup);
+              connectItem.setIcon(
+                  setup ? R.drawable.ic_home_connect_enabled : R.drawable.ic_home_connect_disabled);
+              launchItem.setIcon(
+                  setup ? R.drawable.ic_home_launch_enabled : R.drawable.ic_home_launch_disabled);
             });
-  }
-
-  private void subscribeSocialMediaEvents() {
-    MenuItem socialMediaItem = navbar.getMenu().getItem(SOCIAL_MEDIA_POSITION);
 
-    // Subscribe to lao adding event to adapt the social media menu item
-    mViewModel
-        .getLAOs()
+    // Gray out the social media button if no laos were created
+    viewModel
+        .isSocialMediaEnabled()
         .observe(
             this,
-            laos -> {
-              if (laos.isEmpty()) {
-                socialMediaItem.setIcon(R.drawable.ic_common_social_media_transparent_foreground);
-              } else {
-                socialMediaItem.setIcon(R.drawable.ic_common_social_media_opaque_foreground);
-              }
+            enabled -> {
+              // We set the button icon depending on the livedata value
+              socialMediaItem.setIcon(
+                  Boolean.TRUE.equals(enabled)
```

Comment:
same here and there, can't we use the boolean value ? 😢

Reply:
sadly, no 😞